### ISSUE TITLE
False positive `deprecated-module` caused by name collision

### ISSUE THREAD

**Issue by akosthekiss** [PyCQA/pylint#4629]
### Steps to reproduce

Given multiple files:
```
mypkg/
    formatter/
        __init__.py
    other/
        __init__.py
    __init__.py
```

Where:
```py
"""mypkg/formatter/__init__.py"""

def fmtr():
    """fmtr()"""
```

```py
"""mypkg/other/__init__.py"""

from ..formatter import fmtr
```

```py
"""mypkg/__init__.py"""

from . import formatter
from .formatter import fmtr
```

### Current behavior

Result of `pylint mypkg`:
```
************* Module mypkg
mypkg/__init__.py:4:0: W0402: Uses of a deprecated module 'formatter' (deprecated-module)
************* Module mypkg.other
mypkg/other/__init__.py:3:0: W0402: Uses of a deprecated module 'formatter' (deprecated-module)
```

### Expected behavior

No deprecated-module diagnostics.

### pylint --version output

Result of `pylint --version` output:

```
pylint 3.0.0-a4
astroid 2.6.1
Python 3.6.9 (default, Jan 26 2021, 15:33:00) 
[GCC 8.4.0]
```

```
pylint 2.9.0
astroid 2.6.1
Python 3.6.9 (default, Jan 26 2021, 15:33:00) 
[GCC 8.4.0]
```

**Comment by Pierre-Sassoulas** [comment]
@matusvalo do you have an idea how to fix this ? Can we use the absolute name to do the check ?

**Comment by matusvalo** [comment]
I suppose that yes. I will investigate.

**Comment by matusvalo** [comment]
Please assign this issue to me.

**Comment by Pierre-Sassoulas** [comment]
Thank you for handling this, much appreciated :)
